src/interceptors/axios/utils/index.ts

```diff
@@ -9,3 +19,113 @@ export const getPublicKeyFromContainer = async (containerUrl: string): Promise<s
 
   return data.publicKey;
 };
+
+/**
+ * A Transform stream that encrypts data using the Polaris SDK.
+ */
+export class EncryptStream extends Transform {
+  private sdk: PolarisSDK;
+  constructor(sdk: PolarisSDK) {
+    super();
+    this.sdk = sdk;
+  }
+
+  /**
+   * Encrypts the given data.
+   * @param data - The data to encrypt as a string.
+   * @returns A promise that resolves to the encrypted data as a Buffer.
+   * @throws An error if encryption fails.
+   */
+  async encrypt(data: string): Promise<Buffer> {
+    try {
+      const publicKey = await this.sdk.getPublicKey();
+      return this.sdk.encrypt(Buffer.from(data), publicKey);
+    } catch (error: any) {
+      throw new Error(`Failed to encrypt data: ${error.message}`);
+    }
+  }
+
+  /**
+   * Transforms the input chunk by encrypting it.
+   * @param chunk - The input data chunk as a Buffer.
+   * @param encoding - The encoding type.
+   * @param callback - The callback function to signal completion.
+   */
+  async _transform(chunk: Buffer, encoding: string, callback: Function) {
+    try {
+      const encryptedChunk = await this.encrypt(chunk.toString());
+      this.push(encryptedChunk);
+      callback();
+    } catch (error) {
+      callback(error);
+    }
+  }
+}
+
+/**
+ * A Transform stream that decrypts data using the Polaris SDK.
+ */
+export class DecryptStream extends Transform {
+  private sdk: PolarisSDK;
+  constructor(sdk: PolarisSDK) {
+    super();
+    this.sdk = sdk;
+  }
+
+  /**
+   * Transforms the input chunk by decrypting it.
+   * @param chunk - The input data chunk as a Buffer.
+   * @param encoding - The encoding type.
+   * @param callback - The callback function to signal completion.
+   */
+  async _transform(chunk: Buffer, encoding: string, callback: Function) {
+    try {
+      const decryptedChunk = await this.sdk.decrypt(chunk);
```

Comment:
This may actually be very slow as well. If decrypt needs to do an HTTP request, it will be an issue doing this for every chunk.

I think we need to implement the streaming in a slightly different way. So, when we are creating a new encrypted stream we should:
1. Before the first actual chunk is processed, we create new AES encryption key and IV, wrap them and send them as the first chunk
2. After that we start encrypting the real chunks, but we don't do the full process of creating a new AES key, but reuse the one from the beginning

When decrypting, we do the opposite:
1. Take the wrapped AES key and IV from the first chunk, unwrap them and cache them
2. For each chunk after that, we use the cached AES key and IV to do the symmetric decryption

This has several advantages:
1. An HTTP call needs to do only once for the whole stream (in the beginning)
2. We transfer much less data as each chunk only has the ciphertext and not the "header"